src/BenchmarkDotNet/ConsoleArguments/ConfigParser.cs

```diff
@@ -410,11 +408,21 @@ private static Job CreateJobForGivenRuntime(Job baseJob, string runtimeId, Comma
                     return MakeMonoAOTLLVMJob(baseJob, options, "net6.0");
                 case RuntimeMoniker.MonoAOTLLVMNet70:
                     return MakeMonoAOTLLVMJob(baseJob, options, "net7.0");
+                case RuntimeMoniker.Mono70:
+                    return CreateCoreJob(baseJob, "net7.0", "Mono with .NET 7.0", options, CoreRuntime.Core70)
+                        .WithArguments(new Argument[] { new MsBuildArgument("/p:UseMonoRuntime=true") });

```

Comment:
I am afraid that this might not be enough. I've failed to find any official documentation (this is sad), but according to https://twitter.com/lambdageek/status/1457784508371980291 we need to specify `/p:UseMonoRuntime=true` and publish a self-contained application for specific runtime. 

Here is a small repro:

```cs
Console.WriteLine(Type.GetType("Mono.RuntimeStructs") != null);
```

```log
PS D:\projects\repros\consoleBdn> dotnet publish -c Release -f net7.0 --self-contained -r win-x64 /p:UseMonoRuntime=true
MSBuild version 17.4.0-preview-22428-01+14c24b2d3 for .NET
  Determining projects to restore...
  Restored D:\projects\repros\consoleBdn\consoleBdn.csproj (in 11 ms).
C:\Program Files\dotnet\sdk\7.0.100-rc.1.22431.12\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(219,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-p
olicy [D:\projects\repros\consoleBdn\consoleBdn.csproj]
  consoleBdn -> D:\projects\repros\consoleBdn\bin\Release\net7.0\win-x64\consoleBdn.dll
  consoleBdn -> D:\projects\repros\consoleBdn\bin\Release\net7.0\win-x64\publish\
PS D:\projects\repros\consoleBdn> D:\projects\repros\consoleBdn\bin\Release\net7.0\win-x64\publish\consoleBdn.exe
True
PS D:\projects\repros\consoleBdn> dotnet build -c Release -f net7.0 /p:UseMonoRuntime=true
MSBuild version 17.4.0-preview-22428-01+14c24b2d3 for .NET
  Determining projects to restore...
  Restored D:\projects\repros\consoleBdn\consoleBdn.csproj (in 5 ms).
C:\Program Files\dotnet\sdk\7.0.100-rc.1.22431.12\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(219,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-p
olicy [D:\projects\repros\consoleBdn\consoleBdn.csproj]
  consoleBdn -> D:\projects\repros\consoleBdn\bin\Release\net7.0\consoleBdn.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.29
PS D:\projects\repros\consoleBdn> dotnet D:\projects\repros\consoleBdn\bin\Release\net7.0\consoleBdn.dll
False
```